libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.cpp

```diff
@@ -44,7 +51,7 @@ void AP_BattMonitor_SMBus_Generic::timer()
         return;
     }
 
-    uint16_t data;
+    uint16_t data = UINT16_MAX;
```

Comment:
Why do you need to do this? Data is only ever valid if a read_word() has returned true, and if it hasn't you shouldn't ever look at it, or assume it wasn't changed. This change should only be a slightly larger/slower binary (or you're trusting the compiler to optimize it back out, but either way I can't see the basis for the change)

Reply:
yes, there's no need, removed.